lib/java_buildpack/framework/fusion_reactor.rb

```diff
@@ -0,0 +1,123 @@
+# Cloud Foundry Java Buildpack
+# Copyright 2017 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/framework'
+
+module JavaBuildpack
+  module Framework
+    # Encapsulates the functionality for enabling FusionReactor support.
+    class FusionReactor < JavaBuildpack::Component::VersionedDependencyComponent
+
+      def initialize(context, &version_validator)
+        @component_name                                  = 'FusionReactor'
+        @application                                     = context[:application]
+        @configuration                                   = context[:configuration]
+        @droplet                                         = context[:droplet]
+        return unless supports?
+        @version = 'latest'
+        @uri, @lib_uri, @password, @with_debug = find_agent
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        download_jar self.jar_name, @droplet.sandbox, @component_name
+        return unless @with_debug
+
+        download(@version, @lib_uri, lib_name) do |file|
```

Comment:
We'd prefer a single artifact to be downloaded and exploded in order to ensure that the versions match correctly at all times; effectively we'd like an "atomic" dependency.